src/otx/algorithms/classification/adapters/mmcls/task.py

```diff
@@ -675,23 +678,15 @@ def patch_input_preprocessing(deploy_cfg):
                 mo_options.flags = list(set(mo_options.flags))
 
             def patch_input_shape(deploy_cfg):
-                resize_cfg = get_configs_by_pairs(
-                    cfg.data.test.pipeline,
-                    dict(type="Resize"),
-                )
-                assert len(resize_cfg) == 1
-                resize_cfg = resize_cfg[0]
-                size = resize_cfg.size
-                if isinstance(size, int):
-                    size = (size, size)
+                input_size_manager = InputSizeManager(cfg.data)
+                size = input_size_manager.get_input_size_from_cfg("test")
                 assert all(isinstance(i, int) and i > 0 for i in size)
                 # default is static shape to prevent an unexpected error
                 # when converting to OpenVINO IR
                 deploy_cfg.backend_config.model_inputs = [ConfigDict(opt_shapes=ConfigDict(input=[1, 3, *size]))]
 
             patch_input_preprocessing(deploy_cfg)
-            if not deploy_cfg.backend_config.get("model_inputs", []):
-                patch_input_shape(deploy_cfg)
+            patch_input_shape(deploy_cfg)
```

Comment:
How about integrating `patch_input_shape` into the `patch_input_preprocessing`? I think `preprocessing` include the concept of the shape

Reply:
I agree in some point but there is ambiguous part. `patch_input_preprocessing` changes only data pipeline but `patch_input_shape` changes input size in model part. In my thought, the function name `patch_input_preprocessing` is a little confusing and maybe something like `patch_data_pipeline` is more readable. Anyway changing this code flow seems out of range. How about changing it in other PR after some conversation?

Reply:
Sounds good :)